rust/src/mqtt/detect.rs

```diff
@@ -464,19 +464,61 @@ unsafe extern "C" fn mqtt_reason_code_match(
     tx: *mut c_void, _sig: *const Signature, ctx: *const SigMatchCtx,
 ) -> c_int {
     let tx = cast_pointer!(tx, MQTTTransaction);
-    let ctx = cast_pointer!(ctx, DetectUintData<u8>);
+    let ctx = cast_pointer!(ctx, DetectUintArrayData<u8>);
+    if ctx.index != DetectUintIndex::Any {
+        let mut vals = Vec::new();
```

Comment:
Do we know how many reason codes can we have in a tx? What's the max cost of creating this vector?